packages/lib/server/getLuckyUser.ts

```diff
@@ -1056,16 +1056,15 @@ function getAverageAttributeWeights<
           );
 
           allRRHosts.forEach((rrHost) => {
+            //assignedUser can be undefined if fallback route is hit or in the case of crm ownership
             const assignedUser = attributeOptionWithUsers?.assignedUsers.find(
               (assignedUser) => rrHost.user.id === assignedUser.member.userId
             );
 
-            if (assignedUser) {
-              if (allRRHostsWeights.has(rrHost.user.id)) {
-                allRRHostsWeights.get(rrHost.user.id)?.push(assignedUser.weight ?? 100);
-              } else {
-                allRRHostsWeights.set(rrHost.user.id, [assignedUser.weight ?? 100]);
-              }
+            if (allRRHostsWeights.has(rrHost.user.id)) {
+              allRRHostsWeights.get(rrHost.user.id)?.push(assignedUser?.weight ?? rrHost.weight ?? 100);
```

Comment:
Added the host weight as the fallback. If that's also not available, use 100 as the default